### ISSUE TITLE
API Parse for hybridConnections is case sensitive, does not match reply from Az.

### ISSUE THREAD

**Issue by hoivikaj** [hashicorp/terraform-provider-azurerm#16619]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.1.9

### AzureRM Provider Version

3.4.0

### Affected Resource(s)/Data Source(s)

azurerm_web_app_hybrid_connection

### Terraform Configuration Files

```hcl
I don't believe a configuration file is required to repair this issue, see output.
```

### Debug Output/Panic Output

```shell
│ Error: parsing "/subscriptions/***/resourceGroups/rg/providers/Microsoft.Relay/namespaces/rns/HybridConnections/sql2014-un1-t01": parsing segment "hybridConnections": expected the segment "HybridConnections" to be "hybridConnections"
│
│ parsing "/subscriptions/***/resourceGroups/rg/providers/Microsoft.Relay/namespaces/rns/HybridConnections/sql2014-un1-t01": parsing segment
│ "hybridConnections": expected the segment "HybridConnections" to be "hybridConnections"
```

### Expected Behaviour

Parsing API response from AZ appears to be case sensitive.

### Steps to Reproduce

Azure App Service created, has hybrid connection configured. Ran a terraform apply, and it failed on the refresh portion.

**Comment by myc2h6o** [comment]
This looks duplicate with #20756, which has been fixed in v3.51.0

**Comment by tombuildsstuff** [comment]
Closing since per the above comment this has been fixed, if this is still an issue in the latest version of the Provider then please let us know and we can take another look.

Thanks!
